src/EA.Weee.Web/Areas/Admin/Controllers/ProducerSubmissionController.cs

```diff
@@ -1,13 +1,143 @@
+
+            var model = await serv.Submissions(year);
+
+            model.RegistrationNumber = registrationNumber;
+
+            return View("Producer/ViewOrganisation/OrganisationDetails", model);
+        }
+
+        [AdminSmallProducerSubmissionContext]
+        [HttpGet]
+        public async Task<ActionResult> OrganisationDetails(string registrationNumber, int? year = null)
+        {
+            var serv = submissionService.WithSubmissionData(this.SmallProducerSubmissionData, true);
+
+            var model = await serv.OrganisationDetails(year);
+
+            model.RegistrationNumber = registrationNumber;
+
+            return View("Producer/ViewOrganisation/OrganisationDetails", model);
+        }
+
+        [AdminSmallProducerSubmissionContext]
+        [HttpGet]
+        public async Task<ActionResult> ContactDetails(string registrationNumber, int? year = null)
+        {
+            var serv = submissionService.WithSubmissionData(this.SmallProducerSubmissionData, true);
+
+            var model = await serv.ContactDetails(year);
+
+            model.RegistrationNumber = registrationNumber;
+
+            return View("Producer/ViewOrganisation/ContactDetails", model);
+        }
+
+        [AdminSmallProducerSubmissionContext]
+        [HttpGet]
+        public async Task<ActionResult> ServiceOfNoticeDetails(string registrationNumber, int? year = null)
+        {
+            var serv = submissionService.WithSubmissionData(this.SmallProducerSubmissionData, true);
+
+            var model = await serv.ServiceOfNoticeDetails(year);
+
+            model.RegistrationNumber = registrationNumber;
+
+            return View("Producer/ViewOrganisation/ServiceOfNoticeDetails", model);
+        }
+
+        [AdminSmallProducerSubmissionContext]
+        [HttpGet]
+        public async Task<ActionResult> RepresentedOrganisationDetails(string registrationNumber, int? year = null)
+        {
+            var serv = submissionService.WithSubmissionData(this.SmallProducerSubmissionData, true);
+
+            var model = await serv.RepresentedOrganisationDetails(year);
+
+            model.RegistrationNumber = registrationNumber;
+
+            return View("Producer/ViewOrganisation/RepresentedOrganisationDetails", model);
+        }
+
+        [AdminSmallProducerSubmissionContext]
+        [HttpGet]
+        public async Task<ActionResult> TotalEEEDetails(string registrationNumber, int? year = null)
+        {
+            var serv = submissionService.WithSubmissionData(this.SmallProducerSubmissionData, true);
+
+            var model = await serv.TotalEEEDetails(year);
+
+            model.RegistrationNumber = registrationNumber;
+
+            return View("Producer/ViewOrganisation/TotalEEEDetails", model);
+        }
+
+        [HttpGet]
+        public async Task<ActionResult> AddPaymentDetails()
         {
             return View();
         }
+
+        [HttpGet]
+        public async Task<ActionResult> RemoveSubmission()
```

Comment:
@seanciv Wont need to worry about this one right now but for info the Remove will probably need to hook into the Admin.ProducersController.ConfirmRemoval action. Which means we probably need to return the registeredProducerId from the handler...

Reply:
@JohnPadgettCivica I am going to leave this here because AC states: 
"When I click the 'Remove [YEAR] submission' link
Then I am taken to the Remove submission screen (#77590)"
Which first takes the user to a new screen